src/java/org/apache/cassandra/schema/TableMetadata.java

```diff
@@ -141,6 +147,15 @@ private TableMetadata(Builder builder)
         params = builder.params.build();
         isView = builder.isView;
 
+        isVirtual = builder.isVirtual;
+        if (isVirtual)
+            virtualKlass = builder.virtualKlass;
+        else
+            virtualKlass = null;
+
+        assert isVirtual == flags.contains(Flag.VIRTUAL) || !Strings.isNullOrEmpty(virtualKlass) && !flags.contains(Flag.VIRTUAL)
```

Comment:
Same as above w. the asserts. Let's just draw a line in the sand and stop doing this.